modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/passthru/config/PassThroughConfiguration.java

```diff
@@ -143,11 +145,35 @@ public String getPreserveHttpHeaders() {
     }
 
     public int getConnectionIdleTime() {
-        return getIntProperty(PassThroughConfigPNames.CONNECTION_IDLE_TIME, Integer.MAX_VALUE);
+
+        int idleTime;
+        // Giving higher priority for grace time if it is configured, than for the configured idle time
+        if (isIntPropertyConfigured(PassThroughConfigPNames.CONNECTION_GRACE_TIME)) {
+            idleTime = getDefaultConnectionIdleTime();
```

Comment:
Method name says default, but we are considering the configured value. So method name should be generic, but we can parse a value, that value may be default of configured.

Reply:
We can rename this as getIdleTimeFromgraceTime